src/lua_details.cc

```diff
@@ -114,21 +114,39 @@ void check_script_timeout(lua_State* lua_state)
 std::variant<sol::object, std::string>
 execute_lua_script_safely(sol::state& lua, sol::string_view script)
 {
+    const std::string anchor = u8"\u2693";
+    constexpr gul14::string_view chunk_prefix{ u8"[string \"\u2693\"]:" };
+
+    auto process_msg =
+        [chunk_prefix](gul14::string_view msg) -> std::string
+        {
+            // If C++ code is called by Lua and throws an exception that is not derived
+            // from std::exception, the exception is not intercepted by the Sol
+            // trampoline, but caught directly by Lua. Lua would expect this exception
+            // to come from lua_error() and therefore looks for an error message on its
+            // stack, which is not there. We simply convert this into a more concise
+            // error message.
+            if (msg == "lua: error: stack index 1, expected string, received function")
+                return "Unknown C++ exception";
```

Comment:
This is not triggered on my machine :thinking:, rather Lua returns an empty error message. I'm not sure why.